homeassistant/components/switchbot_cloud/sensor.py

```diff
@@ -52,10 +87,18 @@ async def async_setup_entry(
     """Set up SwitchBot Cloud entry."""
     data: SwitchbotCloudData = hass.data[DOMAIN][config.entry_id]
 
+    async_add_entities(
+        SwitchBotCloudSensor(data.api, device, coordinator, description)
+        for device, coordinator in data.devices.sensors
+        for description in METER_PRO_CO2_SENSOR_DESCRIPTIONS
+        if device.device_type == "MeterPro(CO2)"
+    )
+
     async_add_entities(
         SwitchBotCloudSensor(data.api, device, coordinator, description)
         for device, coordinator in data.devices.sensors
         for description in METER_PLUS_SENSOR_DESCRIPTIONS
+        if device.device_type != "MeterPro(CO2)"
     )
```

Comment:
We should only call async add entities once. I think the most elegant would be if we only have one list comprehension in total

Reply:
Made it one list comprehension, let me know if that's what you were thinking.